src/states/common/base.rs

```diff
@@ -241,19 +245,8 @@ pub trait Base: Display {
     ) -> Transition {
         trace!("{} - ConnectionFailure from {}", self, peer_addr);
 
-        let mut transition = Transition::Stay;
-
-        let pub_ids = self.peer_map_mut().disconnect(peer_addr);
-        for pub_id in pub_ids {
-            trace!("{} - ConnectionFailure from {}", self, pub_id);
-            let other_transition = self.handle_peer_lost(pub_id, outbox);
-
-            if let Transition::Stay = transition {
-                transition = other_transition
-            }
-        }
-
-        transition
+        let _ = self.peer_map_mut().disconnect(peer_addr);
+        self.handle_peer_lost(peer_addr, outbox)
```

Comment:
That is interesting, why this change?

Reply:
We don't have the 1-to-many mapping as when the key was `PublicId`

Reply:
Could you be a bit more verbose, I still don't understand why it is ok to call handle_peer_lost here.

Reply:
Hmm not sure I understand, we called it before as well?

Reply:
Good point, I just completely misread the original loop.